.editorconfig

```
@@ -12,12 +12,14 @@ insert_final_newline = true
 charset = utf-8
 trim_trailing_whitespace = true
 
-# Don't change test output files, pngs or test data files
-[*.{out,png,data}]
+# Don't change test outputs, data files, C files, markdowns and generated files
+[{*.{out,png,data,md,guess,c,h,lock},LICENSE,configure}]
```

Why ignore c and h files?
```suggestion
[{*.{out,png,data,md,guess,c,h,lock},LICENSE,configure}]
```

---

They are covered by citus_indent and I could not come up with editorconfig rules that citus_indent is happy with.

---

I think it's fine not to run eclint check/fix for c and h files if it conflicts with citus_indent. But there's two other main usages for editorconfig matching something that's close to our styling:
1. Editors (including github) will make the code look good by default because they will use 4-width tabs
2. Your editor won't introduce spaces when it should introduce tabs. citus_indent will fix those, but it's nice if your editor is configured correctly for the most basic case.


PS. when I run eclint locally on master it only conflicts with the current indentation in shard_rebalancer.h